### PR TITLE
BigQuery

### PR DESCRIPTION
**Pull request by itsmekumari**
@sawantpritam Please review the BigQuery changes.

### CODE REVIEW

**Comment by sawantpritam** [comment]
Please add proper PR title and proper commit message.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -171,6 +180,84 @@ public static void createTempPartitionedSourceBQTable() throws IOException, Inte
     BeforeActions.scenario.write("BQ Source Table " + bqSourceTable + " created successfully");
   }
 
+  @Before(order = 1, value = "@BQ_SOURCE_DATATYPE_TEST")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Add java doc comment - with list of datatypes

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySource.java [hunk 2]
```diff
@@ -126,4 +126,24 @@ public void validateBigQuerySourceIncorrectPropertyErrorFor(String property, Str
...
     Assert.assertEquals(expectedColor, actualColor);
   }
+
+  @Then("Toggle BigQuery source property enable querying views to true")
+  public void toggleBigQuerySourcePropertyEnableQueryingViewsToTrue() {
+    CdfBigQueryPropertiesActions.toggleEnableQueryingViews();
+  }
+
+  @Then("Enter the BigQuery source property for view materialization project {string}")
+  public void enterTheBigQuerySourcePropertyForViewMaterializationProject(String projectId) {
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySource.java]: projectId -> viewMaterializationProject

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySource.java [hunk 3]
```diff
@@ -126,4 +126,24 @@ public void validateBigQuerySourceIncorrectPropertyErrorFor(String property, Str
...
+    CdfBigQueryPropertiesActions.enterViewMaterializationProject(PluginPropertyUtils.pluginProp(projectId));
+  }
+
+  @Then("Enter the BigQuery source property for view materialization dataset {string}")
+  public void enterTheBigQuerySourcePropertyForViewMaterializationDataset(String dataset) {
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySource.java]: dataset -> viewMaterializationDataset

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java [hunk 4]
```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
...
+  public void toggleBigQuerySinkPropertyRequirePartitionFilterToTrue() {
+    CdfBigQueryPropertiesActions.toggleRequirePartitionFilter();
+  }
+
+  @Then("Enter BigQuery sink property GCS upload request chunk size {string}")
+  public void enterBigQuerySinkPropertyGCSUploadRequestChunkSize(String chunkSize) {
+    CdfBigQueryPropertiesActions.enterChunkSize(PluginPropertyUtils.pluginProp(chunkSize));
+  }
+
+    @Then("Verify the validation error message for invalid chunk size")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java]: Create parameterised common step definition in BigQueryBase-
Verify the BigQuery validation error message for invalid property {string}

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java [hunk 5]
```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
...
+      String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+      Assert.assertEquals(expectedColor, actualColor);
+    }
+
+  @Then("Enter the BigQuery sink properties with incorrect Dataset {string}")
+  public void enterTheBigQuerySinkPropertiesWithIncorrectPropertyValue(String dataset) {
+    CdfBigQueryPropertiesActions.enterBigQueryDataset(PluginPropertyUtils.pluginProp(dataset));
+  }
+
+  @Then("Verify the validation error message for invalid dataset")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java]: Utilise common step definition

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java [hunk 6]
```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
...
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Enter the BigQuery sink properties with incorrect table {string}")
+  public void enterTheBigQuerySinkPropertiesWithIncorrectTable(String table) {
+    CdfBigQueryPropertiesActions.enterBigQueryTable(PluginPropertyUtils.pluginProp(table));
+  }
+
+  @Then("Verify the validation error message for invalid table")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java]: Utilise common step definition

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java [hunk 7]
```diff
@@ -168,4 +171,46 @@ public void validateTheCmekKeyOfTargetBigQueryTableIfCmekIsEnabled(String cmek)
...
+    }
+    BeforeActions.scenario.write("Is Partitioning done on column :" + isPartitioningDoneOnField);
+    Assert.assertEquals("yes", isPartitioningDoneOnField.toLowerCase());
+  }
+
+  @Then("Enter BigQuery property temporary bucket name {string}")
+  public void enterBigQuerySinkPropertyTemporaryBucketName(String temporaryBucket) throws IOException {
+    CdfBigQueryPropertiesActions.enterTemporaryBucketName(PluginPropertyUtils.pluginProp(temporaryBucket));
+  }
+  @Then("Verify the validation error message for invalid temporary bucket")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: Utilise common step definition

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java [hunk 8]
```diff
@@ -168,4 +171,46 @@ public void validateTheCmekKeyOfTargetBigQueryTableIfCmekIsEnabled(String cmek)
       BeforeActions.scenario.write("CMEK not enabled");
     }
   }
+
+  /**
+   * Workaround till https://cdap.atlassian.net/browse/CDAP-18862 gets fixed.
+   * Remove once issue is fixed
+   */
+  @And("Wait for page to render properly")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: Remove.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java [hunk 9]
```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
...
+      String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_CHUNKSIZE);
+      String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("gcsChunkSize").getText();
+      Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+      String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.
+              findPropertyErrorElement("gcsChunkSize"));
+      String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+      Assert.assertEquals(expectedColor, actualColor);
+    }
+
+  @Then("Enter the BigQuery sink properties with incorrect Dataset {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java]: Remove.
Use existing- Enter BigQuery property dataset

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java [hunk 10]
```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
...
+    String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_DATASET);
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("dataset").getText();
+    Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("dataset"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Enter the BigQuery sink properties with incorrect table {string}")
+  public void enterTheBigQuerySinkPropertiesWithIncorrectTable(String table) {
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java]: Remove.
Add common definition in BigQueryBase - Enter BigQuery property table {string}

File: src/e2e-test/features/bigquery/sink/BigQuerySinkError.feature [hunk 11]
```diff
@@ -12,3 +12,49 @@ Feature: BigQuery sink - Validate BigQuery sink plugin error scenarios
...
+    Given Open Datafusion Project to configure pipeline
+    When Sink is BigQuery
+    Then Open BigQuery sink properties
+    Then Enter BigQuery property reference name
+    Then Enter BigQuery property projectId "projectId"
+    Then Enter BigQuery property datasetProjectId "projectId"
+    Then Enter BigQuery property dataset "dataset"
+    Then Enter BigQuery sink property table name
+    Then Enter BigQuery property temporary bucket name "bqInvalidTemporaryBucket"
+    Then Verify the validation error message for invalid temporary bucket
```
Comment (sawantpritam) [inline src/e2e-test/features/bigquery/sink/BigQuerySinkError.feature]: missing newline at the end of file.

File: src/e2e-test/features/bigquery/source/BigQueryToBigQuery.feature [hunk 12]
```diff
@@ -120,3 +120,40 @@ Feature: BigQuery source - Verification of BigQuery to BigQuery successful data
...
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Verify the preview of pipeline is "success"
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Verify the partition table is created with partitioned on field "bqPartitionFieldTime"
```
Comment (sawantpritam) [inline src/e2e-test/features/bigquery/source/BigQueryToBigQuery.feature]: missing newline at the end of file.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java [hunk 13]
```diff
@@ -33,6 +35,7 @@
 import java.io.IOException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.Locale;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: remove

File: src/e2e-test/resources/errorMessage.properties [hunk 14]
```diff
@@ -13,3 +13,7 @@ errorMessagePubSubMaxBatchSizeField=Invalid maximum batch size. Ensure the value
 errorMessagePubSubDelayThresholdField=Invalid delay threshold for publishing a batch. Ensure the value is a positive number.
 errorMessagePubSubRetryTimeout=Invalid max retry timeout for retrying failed publish. Ensure the value is a positive number.
 errorMessagePubSubErrorThreshold=Invalid error threshold for publishing. Ensure the value is a positive number.
+errorMessageIncorrectBQChunkSize=Value must be a multiple of 262144.
+errorMessageIncorrectBQSinkDataset=Dataset name can only contain letters (lower or uppercase), numbers and '_'.
```
Comment (sawantpritam) [inline src/e2e-test/resources/errorMessage.properties]: Create common error message -
errorMessageIncorrectBQProperty=PROPERTY name can only contain letters (lower or uppercase), numbers and '_'.

File: src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java [hunk 15]
```diff
@@ -13,4 +13,8 @@
   public static final String ERROR_MSG_PUBSUB_DELAY_THRESHOLD = "errorMessagePubSubDelayThresholdField";
   public static final String ERROR_MSG_PUBSUB_RETRY_TIMEOUT = "errorMessagePubSubRetryTimeout";
   public static final String ERROR_MSG_PUBSUB_ERROR_THRESHOLD = "errorMessagePubSubErrorThreshold";
+  public static final String ERROR_MSG_BQ_INCORRECT_CHUNKSIZE = "errorMessageIncorrectBQChunkSize";
+  public static final String ERROR_MSG_BQ_INCORRECT_DATASET = "errorMessageIncorrectBQSinkDataset";
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/utils/E2ETestConstants.java]: common ERROR_MSG_BQ_INCORRECT_PROPERTY=errorMessageIncorrectBQProperty

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (itsmekumari) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Added the java doc comment.

**Review by sawantpritam — APPROVED** [review]

**Comment by sawantpritam** [comment]
Please resolve conflicts and squash the commits.

**Pull request closed without merging**